sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/ErrorAttachments.java

```diff
@@ -22,52 +25,87 @@
+     * Build an error attachment log with text suitable for using in {link CrashesListener#getErrorAttachments(ErrorReport)}.
      *
-     * @param text Text to attach to the error report.
-     * @return error Attachment or null if null text is passed.
+     * @param text     text to attach to attachment log.
+     * @param fileName file name to use in error attachment log.
+     * @return ErrorAttachmentLog or null if null text is passed.
      */
-    public static ErrorAttachment attachmentWithText(String text) {
-        return attachment(text, null, null, null);
+    public static ErrorAttachmentLog attachmentWithText(String text, String fileName) {
+        return attachment(null, text, fileName, CONTENT_TYPE_TEXT_PLAIN);
     }
 
     /**
-     * Build an attachment with binary suitable for using in {link CrashesListener#getErrorAttachment(ErrorReport)}.
+     * Build an error attachment log with binary suitable for using in {link CrashesListener#getErrorAttachments(ErrorReport)}.
      *
-     * @param data        binary data.
-     * @param fileName    file name to use on reports.
-     * @param contentType data MIME type.
-     * @return error attachment or null if null data is passed.
+     * @param data     binary data.
+     * @param fileName file name to use in error attachment log.
+     * @return ErrorAttachmentLog attachment or null if null data is passed.
      */
-    public static ErrorAttachment attachmentWithBinary(byte[] data, String fileName, String contentType) {
-        return attachment(null, data, fileName, contentType);
+    public static ErrorAttachmentLog attachmentWithBinary(byte[] data, String fileName, String contentType) {
+        return attachment(data, null, fileName, contentType);
     }
 
     /**
-     * Build an attachment with text and binary suitable for using in {link CrashesListener#getErrorAttachment(ErrorReport)}.
+     * Build an error attachment log with text and binary suitable for using in {link CrashesListener#getErrorAttachments(ErrorReport)}.
      *
-     * @param text        text data.
      * @param data        binary data.
-     * @param fileName    file name to use on reports for the binary data.
+     * @param text        text to attach to the attachment log.
+     * @param fileName    file name to use in error attachment log.
      * @param contentType binary data MIME type.
-     * @return error attachment or null if text and data are null.
+     * @return ErrorAttachmentLog attachment or null if text and data are null.
      */
-    public static ErrorAttachment attachment(String text, byte[] data, String fileName, String contentType) {
-        if (text == null && data == null) {
-            MobileCenterLog.warn(LOG_TAG, "Null content passed to attachment method, returning null");
+    private static ErrorAttachmentLog attachment(byte[] data, String text, String fileName, String contentType) {
+        if (contentType == null) {
```

Comment:
He will know as you validate right after. And either way its when you look at the logs that you notice that. Plus you will have trouble with code coverage if you validate something in double.